### PR TITLE
feat: @IsTaxID validator

### PR DESCRIPTION
**Pull request by cladius6**
## Description

@IsTaxID validator implemented.

## Checklist

- [x] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x] the pull request targets the *default* branch of the repository (`develop`)
- [x] the code follows the established code style of the repository
  - `npm run prettier:check` passes
  - `npm run lint:check` passes
- [x] tests are added for the changes I made (if any source code was modified)
- [x] documentation added or updated
- [x] I have run the project locally and verified that there are no errors

<!-- fixes #[issue number], fixes #[issue number]

### CODE REVIEW

File: src/decorator/string/IsTaxID.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { ValidationOptions } from '../ValidationOptions';
+import { buildMessage, ValidateBy } from '../common/ValidateBy';
+import isTaxIDValidator from 'validator/lib/isTaxID';
+
+export const IS_TAX_ID = 'isTaxID';
+
+/**
+ * Checks if the string is a credit card.
```
Comment (Clashsoft) [inline src/decorator/string/IsTaxID.ts:8]: credit card → tax ID

File: src/decorator/string/IsTaxID.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * (locale is one of ['bg-BG'],['cs-CZ'],['de-AT'],['de-DE'],['dk-DK'],['el-CY'],['el-GR'],['en-CA'],['en-GB'],['en-IE'],['en-US'],['es-ES'],['et-EE'],['fi-FI'],['fr-BE'],['fr-FR'],['fr-LU'],['hu-HU'],['it-IT'],['lv-LV'],['mt-MT'],['nl-NL'],['pl-PL'],['ro-RO'],['sv-SE']).
+ * If given value is not a string, then it returns false.
+ */
+export function isTaxID(value: unknown, locale?: string): boolean {
+  return typeof value === 'string' && isTaxIDValidator(value, locale || 'en-US');
+}
+
+/**
+ * Checks if the string is a credit card.
```
Comment (Clashsoft) [inline src/decorator/string/IsTaxID.ts:17]: credit card → tax ID

**Review by Clashsoft — CHANGES_REQUESTED** [review]

**Comment by NoNameProvided** [comment]
Hey, thanks for opening the PR, please rebase or cherry-pick this to the latest `develop`.

**Comment by NoNameProvided** [comment]
Superset by https://github.com/typestack/class-validator/pull/1822

**Pull request closed without merging**